src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -346,7 +354,7 @@ private static void ValidateAuthorityUri(string authority, AuthorityType? author
             }
 
             string path = authorityUri.AbsolutePath.Substring(1);
-            if (string.IsNullOrWhiteSpace(path))
+            if (string.IsNullOrWhiteSpace(path) && !isCiamAuthority(authority))
```

Comment:
Avoid reparsing this. I would re-write the entire logic, it's not that complex. The logic validation logic needs to avoid parsing at all costs. I would add it as part of `GetAuthorityType` after `ParseAuthority(string)`